internal/discord/command-handlers.go

```diff
@@ -126,6 +137,145 @@ func (d *Discord) LogCommand(i *discordgo.Interaction) (*discordgo.Message, erro
+		if duration > (maxDuration - totalDuration) {
+			return maxDuration, nil
+		}
+		totalDuration += duration
+	}
+
+	return totalDuration, nil
+}
+
+// mapOptions is a helper function that creates a map out of the arguments used in the slash command
+func mapOptions(i *discordgo.InteractionCreate) map[string]*discordgo.ApplicationCommandInteractionDataOption {
+	options := i.ApplicationCommandData().Options
+	optionMap := make(map[string]*discordgo.ApplicationCommandInteractionDataOption, len(options))
+	for _, opt := range options {
+		optionMap[opt.Name] = opt
+	}
+	return optionMap
+}
+
+// AppendLogMsgDescription appends an existing logMsg with the specified text
+func AppendLogMsgDescription(logMsg *discordgo.Message, s string) {
+	if logMsg != nil {
+		logMsg.Embeds[0].Description += fmt.Sprintf("\n%v", s)
+	}
+}
+
+// EditLogMsg sends the updated logMsg to Discord and overwrites the message referred to by logMsg.ID
+func (d *Discord) EditLogMsg(logMsg *discordgo.Message) {
+	if logMsg != nil {
+		_, err := d.Session.ChannelMessageEditEmbed(d.ModLoggingChannelID, logMsg.ID, logMsg.Embeds[0])
+		if err != nil {
+			fmt.Printf("Unable to edit log message: %v\n", err)
+		}
+	}
+}
+
+// UpdateLogMsgTimestamp updates the timestamp for the embed in a logMsg
+func UpdateLogMsgTimestamp(logMsg *discordgo.Message) {
+	if logMsg != nil {
+		logMsg.Embeds[0].Timestamp = time.Now().Format(time.RFC3339)
+	}
+}
+
+// SendDMToUser sends a DM to the user specified in `userID` with `message` as its contents
+func (d *Discord) SendDMToUser(state *InteractionState, userID string, message string) error {
+	// Open DM channel with user
+	channel, err := state.session.UserChannelCreate(userID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not create a DM with user %v", userID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(state.session, state.interaction.Interaction, tempstr, &state.isFirst)
+		AppendLogMsgDescription(state.logMsg, "Failed to notify user via DM")
+		return err
+	} else {
+		_, err = state.session.ChannelMessageSend(channel.ID, message)
+		if err != nil {
+			tempstr := fmt.Sprintf("Could not send a DM to user <@%v>", userID)
+			fmt.Printf("%v: %v\n", tempstr, err)
+			RespondToInteraction(state.session, state.interaction.Interaction, tempstr, &state.isFirst)
+			AppendLogMsgDescription(state.logMsg, "Failed to notify user via DM")
+			return err
+		}
+		return nil
+	}
+}
+
+// CheckUserForRoles takes a member and a slice of rolesToCheck (role IDs) and returns
+// a map of bools indicating whether or not the role is present
+func CheckUserForRoles(member *discordgo.Member, rolesToCheck []string) map[string]bool {
+	// initialize map
+	presentRoles := make(map[string]bool)
+	for _, roleToCheck := range rolesToCheck {
+		presentRoles[roleToCheck] = false
+	}
+
+	// check whether user has role
+	for _, role := range member.Roles {
+		if slices.Contains(rolesToCheck, role) {
+			presentRoles[role] = true
+		}
+	}
```

Comment:
we can condense this into one for loop and check the user's roles against `rolesToCheck`